gpii/node_modules/flowManager/src/SolutionsRegistryDataSource.js

```diff
@@ -35,45 +43,209 @@ fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource", {
     }
     var url = fluid.module.resolvePath(that.options.path);
     if (!fs.existsSync(url)) {
         fluid.fail("The path provided to the solutionsRegistry datasource (", url, ") has not been found on the file system");
     }
-
     that.fullSolutionsRegistry = fluid.freezeRecursive(require(url));
 };
 
+// Solutions registry data source used by the Cloud Based Flow Manager
+fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource.cloudBased", {
+    gradeNames: ["gpii.flowManager.solutionsRegistry.dataSource"],
+    readOnlyGrade: "gpii.flowManager.solutionsRegistry.dataSource.cloudBased",
+    invokers: {
+        getImpl: {
+            funcName: "gpii.flowManager.solutionsRegistry.dataSource.cloudBased.handle",
+            args: ["{that}", "{arguments}.1", "{arguments}.2"]
+                             // options, directModel
+        }
+    },
+    listeners: {
+        "loadSolutions.loadFromLocalDisk": {
+            listener: "{that}.loadFromLocalDisk",
+            priority: "first"
+        },
+        "loadSolutions.solutionsLoaded": {
+            listener: "{that}.events.solutionsRegistryReady",
+            priority: "after:loadFromLocalDisk"
+        }
+    }
+});
+
 /**
- * Handler for get requests of solutions registry. It will return either a full solution registry,
- * or if an 'os' is provided in the requestOptions, only the entries for that os will be returned
+ * Handler for get requests to the solutions registry used by the Cloud Based
+ * Flow Manager. It will return an object containing the solutions registry for
+ * the given platform.
  *
- * @param {Object} that - The gpii.flowManager.solutionsRegistry.dataSource.
- * @param {Object} requestOptions - Currently the only request option supported is "os". If provided,
- *           the returned solutions registry will be filtered by OS version.
- * @return {Promise} A promise that will be resolved with results (see above) or rejected on error.
+ * @param {Object} that - The gpii.flowManager.solutionsRegistry.dataSource.cloudBased
+            instance that is handling the GET request.
+ * @param {Object} requestOptions - Currently the only request option supported
+ *          is "os". The returned solutions registry will be the solutions for
+ *          the OS version provided.
+ * @return {Promise} A promise that will be resolved with results (see above) or
+ *          rejected on error.
+ */
+gpii.flowManager.solutionsRegistry.dataSource.cloudBased.handle = function (that, requestOptions) {
+    var promise = fluid.promise();
+    if (requestOptions.os && requestOptions.os in that.fullSolutionsRegistry) {
+        promise.resolve(that.fullSolutionsRegistry[requestOptions.os]);
+    } else {
+        promise.reject({
+            isError: true,
+            message: "The requested OS (" + requestOptions.os + ") was not present in the solutions registry",
+            statusCode: 404
```

Comment:
Same as the comment above, don't need to return `statusCode: 404`. Same comment applies to line 254, 261.

Reply:
Okay, but this one (L129) is old code from the original commit back in 2016:
https://github.com/GPII/universal/blob/f1c2eee4c99e06a60a06dcf78147f3bbe22d5f20/gpii/node_modules/flowManager/src/SolutionsRegistryDataSource.js#L75

As far as I can tell, the authors' intent was to simulate a GET request/response paradigm and that paradigm includes status codes.  But, as I [wrote above -- last paragraph](https://github.com/GPII/universal/pull/840#discussion_r395760908), I don't see a reason for doing this since by the time the function is called, the information is available and there is no need to wrap it in a promise.

In any case, I removed the status code.